bundles/org.openhab.binding.wundergroundupdatereceiver/README.md

```diff
@@ -0,0 +1,227 @@
+
+#### Synthetic channel-types. These are programmatically added:
+
+|  Channel type id       | Type                 | Channel type | Label                    | Description                                                                                                                                                | Group    |
+|------------------------|----------------------|--------------|--------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------|----------|
+| dateutc-datetime       | dateutc-datetime     | state        | Last Updated as DateTime | The date and time of the last update in UTC as submitted by the weather station converted to a DateTime value. In case of 'now', the current time is used. | Metadata | 
+| last-received-datetime | DateTime             | state        | Last Received            | The date and time of the last update.                                                                                                                      | Metadata | 
+| last-query-state       | String               | state        | The last query           | The part of the last query after the first unurlencoded '?'                                                                                                | Metadata |
+| last-query-trigger     | String               | trigger      | The last query           | The part of the last query after the first unurlencoded '?'                                                                                                | Metadata |
+
+The trigger channel's payload is the last querystring, so the following dsl rule script would send the measurements on to wunderground.com:
+
+```
+val requestQuery = receivedEvent
+sendHttpGetRequest("https://rtupdate.wunderground.com/weatherstation/updateweatherstation.php?" + requestQuery)
+```
+
+The PASSWORD, action and realtime parameters are ignored, as they are wunderground technical constants, that devices must send.
+The ID parameter is used to identify the correct thing the request pertains to, i.e. the stationId configuration value.
+
+As described by the wunderground specification a device can submit multiple values for the outdoor temperature, soil temperature, soil moisture and leaf wetness channels by inserting an index number into the name of the request parameter, fx. tempf can be temp1f, temp2f, etc.
+This is supported by the discovery mechanism, creating a channel for each of the values.
+
+# Examples
+
+### Thing file
+
+Configuration using thing and item files is not the recommended method, as you have to manually replicate the configuration, discovery produces.
+Items must be named as the request parameters in the channel type table and must have the same types.
```

Comment:
> There are some checkstyle warnings left. You could take a look at `target/code-analysis/report.html` after building it with `mvn clean install`. The following should be fixed:
> 
> ```
> [WARNING] .binding.wundergroundupdatereceiver\README.md:[167]
> Missing an empty line after the Markdown header (#).
> [WARNING] .binding.wundergroundupdatereceiver\README.md:[168]
> The line before code formatting section must be empty.
> [WARNING] org.openhab.binding.wundergroundupdatereceiver.internal.WundergroundUpdateReceiverDiscoveryServiceTest.java:[30]
> The package org.apache.commons.lang3.ArrayUtils should not be used.
> [WARNING] org.openhab.binding.wundergroundupdatereceiver.internal.WundergroundUpdateReceiverDiscoveryServiceTest.java:[58]
> Classes/Interfaces/Enums should be annotated with @NonNullByDefault
> [WARNING] org.openhab.binding.wundergroundupdatereceiver.internal.WundergroundUpdateReceiverServletTest.java:[82]
> Classes/Interfaces/Enums should be annotated with @NonNullByDefault
> ```
> 
> Would be great if you could replace the two array operations by native code as we want to get rid of Apache commons.
> 
> Please run `mvn org.openhab.core.tools:i18n-maven-plugin:3.3.0-SNAPSHOT:generate-default-translations` to generate the default translations. Then, your binding can be translated by other users via Crowdin.
> 
> Today's afternoon (probably) is code freeze for OH 3.3. If you can make it to fix the findings, your binding will be part of the new release. Otherwise, it can be merged after 27th June, after OH 3.3 has been released.

You are correct, it should channels and not items. In connection with the generate-default-translations, the pom.xml is missing a `pluginRepository` block pointing to `https://openhab.jfrog.io/artifactory/libs-snapshot`, which is where the snapshot version of the i18n plugin can be found. It isn't on maven central, so it took me a little duckduckgoing to figure out. Where to post an issue about this?

Reply:
It is already addressed here: https://github.com/openhab/openhab-docs/pull/1833#discussion_r888723306